metadata-ingestion/src/datahub/ingestion/source/metabase.py

```diff
@@ -0,0 +1,332 @@
+
+        self.session = requests.Session()
+        self.session.headers.update(
+            {
+                "X-Metabase-Session": f"{self.access_token}",
+                "Content-Type": "application/json",
+                "Accept": "*/*",
+            }
+        )
+
+        # Test the connection
+        test_response = self.session.get(f"{self.config.connect_uri}/api/user/current")
+        if test_response.status_code == 200:
+            pass
+
+        if test_response.status_code == 401:
+            self.report.report_failure(
+                key="metabase-session",
+                reason=f"Unable to login for user {self.config.username}. Unauthorized access",
+            )
+
+    def close(self) -> None:
+        response = requests.delete(
+            f"{self.config.connect_uri}/api/session",
+            headers={"X-Metabase-Session": self.access_token},
+        )
+        if response.status_code != 204:
+            self.report.report_failure(
+                key="metabase-session",
+                reason=f"Unable to logout for user {self.config.username}",
+            )
+
+    def emit_dashboard_mces(self) -> Iterable[MetadataWorkUnit]:
+        dashboard_response = self.session.get(
+            f"{self.config.connect_uri}/api/dashboard"
+        )
+        payload = dashboard_response.json()
+        for dashboard_info in payload:
+            dashboard_snapshot = self.construct_dashboard_from_api_data(dashboard_info)
+
+            mce = MetadataChangeEvent(proposedSnapshot=dashboard_snapshot)
+            wu = MetadataWorkUnit(id=dashboard_snapshot.urn, mce=mce)
+            self.report.report_workunit(wu)
+
+            yield wu
+
+    def construct_dashboard_from_api_data(
+        self, dashboard_info: dict
+    ) -> DashboardSnapshot:
+        dashboard_url = (
+            f"{self.config.connect_uri}/api/dashboard/{dashboard_info['id']}"
+        )
+        dashboard_response = self.session.get(dashboard_url)
+        dashboard_details = dashboard_response.json()
+        dashboard_urn = f"urn:li:dashboard:({self.platform},{dashboard_details['id']})"
+        dashboard_snapshot = DashboardSnapshot(
+            urn=dashboard_urn,
+            aspects=[],
+        )
+        last_edit_by = dashboard_details.get("last-edit-info") or {}
+        modified_actor = f"urn:li:corpuser:{last_edit_by.get('email', 'unknown')}"
+        modified_ts = int(
+            dp.parse(f"{last_edit_by.get('timestamp', 'now')}").timestamp() * 1000
+        )
+        title = dashboard_details.get("name", "") or ""
+        description = dashboard_details.get("description", "") or ""
+        last_modified = ChangeAuditStamps(
+            created=AuditStamp(time=modified_ts, actor=modified_actor),
+            lastModified=AuditStamp(time=modified_ts, actor=modified_actor),
+        )
+
+        card_urns = []
+        cards_data = dashboard_details.get("ordered_cards", "{}")
```

Comment:
Nit: let's name this chart_urns

Reply:
We should aim to keep Metabase concepts named as they are in Metabase, but once they are mapped to DataHub concepts they should take on DataHub names

Reply:
fixed please see [commit 0c03ae4bc9ca7e473bb7c5c7f0ab6f3a1fb446ab](https://github.com/linkedin/datahub/pull/3602/commits/0c03ae4bc9ca7e473bb7c5c7f0ab6f3a1fb446ab)